misaligned.c

```diff
@@ -1,21 +1,29 @@
-#include <stdio.h>
-#include <assert.h>
+#include "misaligned.h"
 
-int printColorMap() {
-    const char* majorColor[] = {"White", "Red", "Black", "Yellow", "Violet"};
-    const char* minorColor[] = {"Blue", "Orange", "Green", "Brown", "Slate"};
-    int i = 0, j = 0;
+Colorpair colorpair;
+
+int printAndUpdateColorData(int pairnumber) {
+    int i = 0, j = 0, localPairNumber=0;
     for(i = 0; i < 5; i++) {
         for(j = 0; j < 5; j++) {
+            localPairNumber = localPairNumber+1;
+            if (localPairNumber == pairnumber)
+            {
+                strcpy(colorpair.MajorColor , majorColor[i]);
+	            strcpy(colorpair.MinorColor , minorColor[i]);
+            }

```

Comment:
the 'sampling' (of pairnumber) is needed for testing, but it's embedded in the production code.
Can we separate the formatting and printing, so the formatting can be tested (and fixed) on its own?

Reply:
As suggested , the changes have been made and pushed to GIT